### PR TITLE
Add data import cron resource

### PR DESCRIPTION
**Pull request by duyanyan**
##### Short description:
Add data import cron resource

### CODE REVIEW

File: ocp_resources/data_import_cron.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+
+class DataImportCron(NamespacedResource):
+    """
+    DataImportCron object.
+    """
+
+    api_group = NamespacedResource.ApiGroup.CDI_KUBEVIRT_IO
+
+    def __init__(
```
Comment (rnetser) [inline ocp_resources/data_import_cron.py:11]: please add privileged_client=None

File: ocp_resources/data_import_cron.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        self.imports_to_keep = imports_to_keep
+
+    def to_dict(self):
+        res = super().to_dict()
+        if self.yaml_file:
+            return res
+
+        res.update(
+            {
+                "spec": {
```
Comment (rnetser) [inline ocp_resources/data_import_cron.py:50]: `["spec"]["template"]["spec"]` - save and reuse

File: ocp_resources/data_import_cron.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+                    "template": {
+                        "spec": {
+                            "source": {
+                                "registry": {
+                                    "imageStream": self.image_stream,
```
Comment (rnetser) [inline ocp_resources/data_import_cron.py:55]: not necessarily `imageStream`, can also be `url`

File: ocp_resources/data_import_cron.py [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+                                    "pullMethod": self.pull_method,
```
Comment (rnetser) [inline ocp_resources/data_import_cron.py:56]: what about (optinal? )  `certConfigMap`?

**Review by rnetser — CHANGES_REQUESTED** [review]

**Comment by duyanyan** [comment]
@rnetser  please review again

File: ocp_resources/data_import_cron.py [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+                            "source": {"registry": {"pullMethod": self.pull_method}},
+                            "storage": {
+                                "resources": {"requests": {"storage": self.size}}
+                            },
+                        }
+                    }
+                }
+            }
+        )
+        spec = res["spec"]["template"]["spec"]
```
Comment (rnetser) [inline ocp_resources/data_import_cron.py:67]: should be before line 53

File: ocp_resources/data_import_cron.py [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+        if self.image_stream:
```
Comment (rnetser) [inline ocp_resources/data_import_cron.py:68]: it's either `image_stream` or `url`, right? if so, please add a verification that indeed only one is passed. if both - raise

**Review by rnetser — CHANGES_REQUESTED** [review]

File: ocp_resources/data_import_cron.py (hunk 5 above)
Comment (rnetser) [reply ocp_resources/data_import_cron.py:67]: Please ignore and leave it here

File: ocp_resources/data_import_cron.py (hunk 6 above)
Comment (duyanyan) [reply ocp_resources/data_import_cron.py:70]: Fixed, please review, thanks!

File: ocp_resources/data_import_cron.py [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+                                "resources": {"requests": {"storage": self.size}}
+                            },
+                        }
+                    }
+                }
+            }
+        )
+        spec = res["spec"]["template"]["spec"]
+        if self.image_stream and self.url:
+            raise ValueError("imageStream and url can not be coexisted")
```
Comment (rnetser) [inline ocp_resources/data_import_cron.py:69]:
```suggestion
            raise ValueError("imageStream and url can not coexist")
```

Comment (duyanyan) [reply ocp_resources/data_import_cron.py]: Fixed

**Comment by duyanyan** [comment]
@rnetser Could you please review again?

File: ocp_resources/data_import_cron.py [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+                            "storage": {
+                                "resources": {"requests": {"storage": self.size}}
+                            },
+                        }
+                    }
+                }
+            }
+        )
+        spec = res["spec"]["template"]["spec"]
+        if self.image_stream and self.url:
```
Comment (myakove) [inline ocp_resources/data_import_cron.py:68]: Please call it first `if self.image_stream and self.url:`

**Review by myakove — CHANGES_REQUESTED** [review]

File: ocp_resources/data_import_cron.py (hunk 8 above)
Comment (duyanyan) [reply ocp_resources/data_import_cron.py]: @myakove could you please review again?

**Review by myakove — APPROVED** [review]

**Pull request merged**